diff --git a/glib/CMakeLists.txt b/glib/CMakeLists.txt
index 2781f09..77b6490 100644
--- a/glib/CMakeLists.txt
+++ b/glib/CMakeLists.txt
@@ -126,7 +126,7 @@ endmacro(_list_prefix)
 if (HAVE_INTROSPECTION AND BUILD_SHARED_LIBS)
   include(GObjectIntrospectionMacros)
   set(INTROSPECTION_GIRS)
-  set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR} --warn-all")
+  set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR} --warn-all --library-path=${CMAKE_CURRENT_BINARY_DIR} --library-path=${CMAKE_CURRENT_BINARY_DIR}/..")
   set(INTROSPECTION_COMPILER_ARGS "--includedir=${CMAKE_CURRENT_SOURCE_DIR}")
 
   set(introspection_files ${poppler_glib_SRCS} ${poppler_glib_public_headers})
